src/message_sender.c

```diff
@@ -562,7 +562,6 @@ static void on_link_state_changed(void* context, LINK_STATE new_link_state, LINK
         {
             /* User initiated transition, we should be good */
             set_message_sender_state(message_sender_instance, MESSAGE_SENDER_STATE_IDLE);
-            indicate_all_messages_as_error(message_sender_instance);
```

Comment:
@Shakti213 Let me know if you want to push a new commit and I'll have a look and merge it in. Or otherwise I can create a fix that you can have a look at.

---
In reply to: [126278896](https://github.com/Azure/azure-uamqp-c/pull/157#discussion_r126278896) [](ancestors = 126278896)